invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/DocumentRequestForm/DocumentRequestForm.js

```diff
@@ -0,0 +1,211 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { Formik } from 'formik';
+import { Button, Container, Form, Header, Segment } from 'semantic-ui-react';
+import { sessionManager } from '../../../authentication/services';
+import * as Yup from 'yup';
+import { FrontSiteRoutes } from '../../../routes/urls';
+import { Loader } from '../../../common/components';
+import get from 'lodash/get';
+
+const ERROR_MSGS = {
+  publication_year: 'Not a valid year',
+};
+
+const RequestSchema = Yup.object().shape({
+  title: Yup.string().required(),
+  publication_year: Yup.number()
+    .typeError(ERROR_MSGS.publication_year)
+    .integer(ERROR_MSGS.publication_year),
+});
+
+export default class DocumentRequestForm extends Component {
+  state = {
+    data: {},
+  };
+
+  get initialValues() {
+    const values = {
+      title: get(this.props, 'location.state.queryString', ''),
+      authors: '',
+      isbn: '',
+      issn: '',
+      volume: '',
+      issue: '',
+      page: '',
+      publication_year: '',
+      note: '',
+    };
+    for (const [key, value] of Object.entries(this.state.data)) {
```

Comment:
@equadon I am using the lodash `pick` function for that ;) Like that 
```
prepareData = data => {
    return _.pick(data, [
      'abstract',
      'authors',
      'edition',
      'issn',
      'language',
      'mode_of_issuance',
      'title',
    ]);
  };
```

Reply:
Hm, yes it might be possible to use `pick` here. Could also get the values from `this.state.data` and instead of having the `initialValues` we set `state.data` to those values in the constructor.

Reply:
In principle you don't need to pass initial values so also this is an option. You could just do ``data ? data : {}

Reply:
True but I changed it now so the constructor initializes `state.data` which is later updated on submit.